platform/o.n.swing.laf.flatlaf/src/org/netbeans/swing/laf/flatlaf/ui/FlatEditorTabCellRenderer.java

```diff
@@ -219,11 +222,17 @@ private void paintInteriorAtScale1x(Graphics2D g, Component c, int width, int he
             if (selected && underlineHeight > 0) {
                 // paint underline if tab is selected
                 int underlineHeight = (int) Math.round(FlatEditorTabCellRenderer.underlineHeight * scale);
-                g.setColor(ren.isActive() ? underlineColor : inactiveUnderlineColor);
-                if (underlineAtTop)
+                if (underlineAtTop) {
+                    g.setColor(contentBorderColor);
+                    int borderWidth = (int) (1 * scale);
+                    g.fillRect(0, 0, borderWidth, height);
+                    g.fillRect(width - tabSeparatorWidth - borderWidth, 0, borderWidth, height);
+                    g.setColor(ren.isActive() ? underlineColor : inactiveUnderlineColor);
                     g.fillRect(0, 0, width - tabSeparatorWidth, underlineHeight);
```

Comment:
What is this code doing? It seems it's trying to draw side borders if underlineAtTop is true. That seems to be a bit ad hoc, and should probably be in a separate configuration property.

Changes here are going to start conflicting with the changes in https://github.com/apache/netbeans/pull/3115 , which I'm still hoping to integrate later. Perhaps we could keep this PR limited to simple changes to FlatLAF.properties, as was originally intended? I can help make these improvements later...

Reply:
I reverted the border colour change, and it looks odd without this. The "underline" sits on the content border now either way. I think it's needed for consistency, and I'd like this to work standalone if merged. You can enhance or remove as needed then, but this is a more sensible default IMO.

Reply:
OK, as long as we agree that this code might be adjusted in the future.